docs/src/_concepts/state_backends.md

```diff
@@ -0,0 +1,95 @@
+
+
+### Google Cloud Storage
+To store state remotely in Google Cloud Storage, set the `state_backend.uri` setting to `gs://<your bucket name>/<prefix for state JSON blobs>`.
+
+To authenticate to GCS, you must provide a path to a (service account credentials file)[https://cloud.google.com/iam/docs/creating-managing-service-account-keys]
+These can be configured via the `state_backend.gcs.application_credentials` setting.
+
+If credentials are not provided via these settings, Meltano will use the value the `GOOGLE_APPLICATION_CREDENTIALS` environment variable, if it is set.
+
+If GCS credentials are not found via any of the methods described above, Meltano will not be able to authenticate to Google Cloud Storage and state operations will fail.
+
+## Configuration
+
+The state backend settings for your Meltano project can be configured much the same as any other Meltano setting. For some state backends, the only required setting is `state_backend.uri`.
+By default, `state_backend.uri` is set to the special keyword string `systemdb`, which will tell Meltano to store state in the same backend database as it stores other project data.
+
+It can be changed by running `meltano config meltano set state_backend.uri <URI for desired state backend>` or by directly editing a project's `meltano.yml` to add the following:
+```yaml
+state_backend:
+    uri: <URI for desired state backend>
+```
+For some state backends and in some environments, `uri` is the only setting that needs to be configured in order to use a state backend other than the default `systemdb` backend. For instance, to store state directly on the local filesystem at the path `${MELTANO_PROJECT_ROOT}/.meltano/state`, simply run `meltano config meltano set 'state_backend.uri file:///${MELTANO_PROJECT_ROOT}/.meltano/state'`. Note the single quotes which prevent the early expansion of the environment variable.
+Then Meltano will store the state for a given `state_id` at the path `${MELTANO_PROJECT_ROOT}/.meltano/state/<state_id>/state.json`.
+
+### Locking
+
+Because the `systemdb` state backend utilizes a transactional database, it can rely on the database's transaction logic to prevent conflicts among multiple runs of the same pipeline.
+For some other state backends, Meltano implements its own simple locking mechanism.
+This locking mechanism utilizes reasonable defaults but you may wish to configure the locking settings differently.
+You can do this via two additional top-level `state_backend` settings: `state_backend.lock_timeout_seconds` and `state_backend.lock_retry_seconds`.
+
+When Meltano tries to read or write state for a given `state_id`, it will try to acquire a lock on the state data for that `state_id`.
+For example, using the local filesystem state backend with `state_backend.uri` set to `file:///${MELTANO_PROJECT_ROOT}/.meltano/state`, it will check to see if a file exists at the path `${MELTANO_PROJECT_ROOT}/.meltano/state/<state_id>/lock`.
```

Comment:
```suggestion
For example, using the local filesystem state backend with `state_backend.uri` set to `file:///${MELTANO_SYS_DIR_ROOT}/state`, it will check to see if a file exists at the path `file:///${MELTANO_SYS_DIR_ROOT}/state/<state_id>/lock`.
```